src/IO/ReadBuffer.h

```diff
@@ -63,20 +63,23 @@ class ReadBuffer : public BufferBase
         chassert(!hasPendingData());
         chassert(position() <= working_buffer.end());
 
-        bytes += offset();
-        bool res = nextImpl();
-        if (!res)
+        bool res = true;
+        do
         {
-            working_buffer = Buffer(pos, pos);
-        }
-        else
-        {
-            pos = working_buffer.begin() + std::min(nextimpl_working_buffer_offset, working_buffer.size());
-            chassert(position() < working_buffer.end());
-        }
-        nextimpl_working_buffer_offset = 0;
-
-        chassert(position() <= working_buffer.end());
+            bytes += offset();
+            res &= nextImpl();
+            if (!res)
+            {
+                working_buffer = Buffer(pos, pos);
+            }
+            else
+            {
+                pos = working_buffer.begin() + std::min(nextimpl_working_buffer_offset, working_buffer.size());
+            }
+            nextimpl_working_buffer_offset = 0;
+        } while (res && available() == 0);
```

Comment:
I don't think this is a right place to fix this issue - the problem is with ZlibInflatingReadBuffer